x/sequencer/keeper/slashing.go

```diff
@@ -31,20 +31,34 @@ func (k Keeper) Slashing(ctx sdk.Context, seqAddr string) error {
 		k.Logger(ctx).Error("sequencer has no tokens to slash", "sequencer", seq.SequencerAddress)
 	}
 	seq.Tokens = sdk.Coins{}
-
 	oldStatus := seq.Status
-	wasProposer := seq.Proposer
-	// in case we are slashing an unbonding sequencer, we need to remove it from the unbonding queue
+
+	// remove from queue if unbonding
 	if oldStatus == types.Unbonding {
 		k.removeUnbondingSequencer(ctx, seq)
 	}
+	// remove from notice period queue if needed
+	if seq.IsNoticePeriodInProgress() {
+		k.removeNoticePeriodSequencer(ctx, seq)
+	}
+	// if the slashed sequencer is the proposer, remove it
+	// the caller should rotate the proposer
+	if k.isProposer(ctx, seq.RollappId, seqAddr) {
+		k.removeProposer(ctx, seq.RollappId)
+	}
```

Comment:
if the caller needs to rotate, why cant the caller also remove?

Reply:
because having defined proposer while it's unbonded, is invariant breaking.
when unbonding a sequencer, it can't be left as a proposer.
whether rotation required, is handled by the caller